WcaOnRails/app/models/competition.rb

```diff
@@ -368,6 +368,10 @@ def warnings_for(user)
         warnings[:name] = I18n.t('competitions.messages.name_too_long')
       end
 
+      if !/[[:upper:]]/.match(self.id[0,1])
```

Comment:
Nit: you can write this as `unless /^[[:upper:]]/.match(self.id)` (note how we use `^` to match against the start of the string, instead of slicing the matched string).